components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java

```diff
@@ -92,6 +93,297 @@
+                            defaultScopeBindings.put(noneDefaultTypeScope, noneDefaultTypeeRole);
+                        }
+                    }
+                    defaultTypeExtension.put(APIConstants.SWAGGER_X_SCOPES_BINDINGS, defaultScopeBindings);
+                    defaultTypeFlow.setVendorExtensions(defaultTypeExtension);
+                    securityDefinitions.put(SWAGGER_SECURITY_SCHEMA_KEY, defaultTypeFlow);
+
+                }
+            }
+        }
+        swagger.setSecurityDefinitions(securityDefinitions);
+        if (securityDefinitions != null
+                && (oAuth2Definition = (OAuth2Definition) securityDefinitions.get(SWAGGER_SECURITY_SCHEMA_KEY)) != null
+                && oAuth2Definition.getScopes() != null) {
+            for (Map.Entry<String, String> entry : oAuth2Definition.getScopes().entrySet()) {
+                Scope scope = new Scope();
+                scope.setKey(entry.getKey());
+                scope.setName(entry.getKey());
+                scope.setDescription(entry.getValue());
+                Map<String, String> scopeBindings;
+                if (oAuth2Definition.getVendorExtensions() != null && (scopeBindings =
+                        (Map<String, String>) oAuth2Definition.getVendorExtensions()
+                                .get(APIConstants.SWAGGER_X_SCOPES_BINDINGS)) != null) {
+                    if (scopeBindings.get(scope.getKey()) != null) {
+                        scope.setRoles(scopeBindings.get(scope.getKey()));
+                    }
+                }
+                scopeSet.add(scope);
+            }
+            setOtherSchemes(otherSetOfSchemes);
+            return OASParserUtil.sortScopes(scopeSet);
+        } else {
+            setOtherSchemes(otherSetOfSchemes);
+            return OASParserUtil.sortScopes(getScopesFromExtensions(swagger));
+        }
+    }
+
+    /**
+     * This method returns URI templates according to the given swagger file(Swagger version 2)
+     *
+     * @param resourceConfigsJSON swaggerJSON
+     * @return URI Templates
+     * @throws APIManagementException
+     */
+    @Override
+    public Set<URITemplate> injectOtherResourceScopesToDefaultScheme(String resourceConfigsJSON) throws APIManagementException {
+        Swagger swagger = getSwagger(resourceConfigsJSON);
+        Set<URITemplate> urlTemplates = new LinkedHashSet<>();
+        Set<Scope> scopes = injectOtherScopesToDefaultScheme(resourceConfigsJSON);
+        List<String> schemes = getOtherSchemes();
+        boolean isDefaultAvailable = isDefaultGiven(resourceConfigsJSON);
+
+        for (String pathString : swagger.getPaths().keySet()) {
+            Path path = swagger.getPath(pathString);
+            Map<HttpMethod, io.swagger.models.Operation> operationMap = path.getOperationMap();
+            for (Map.Entry<HttpMethod, io.swagger.models.Operation> entry : operationMap.entrySet()) {
+                io.swagger.models.Operation operation = entry.getValue();
```

Comment:
`io.swagger.models.Operation` is already imported to the class. Shall we use the shortened form `Operation` ?